tests/test_new_talk_voting.py

```diff
@@ -2,31 +2,57 @@
 from unittest import mock
 
 import pytest
-from django.conf import settings
 from django.core.urlresolvers import reverse
 from django.utils import timezone
 
-from conference.models import Conference, VotoTalk, TALK_STATUS
+from conference.models import Conference, VotoTalk, TALK_STATUS, TALK_TYPE_CHOICES, TALK_ADMIN_TYPE
 from conference.talk_voting import VotingOptions, find_talks
-from tests.factories import SpeakerFactory, TalkFactory, TalkSpeakerFactory
-from tests.common_tools import make_user, create_talk_for_user, get_default_conference
+from tests.factories import SpeakerFactory, TalkFactory, TalkSpeakerFactory, TicketFactory
+from tests.common_tools import make_user, create_talk_for_user, get_default_conference, template_used
 
 pytestmark = [pytest.mark.django_db]
 
 
-@pytest.mark.xfail
-def test_talk_voting_unavailable_if_not_enabled():
-    assert False
+def test_talk_voting_unavailable_before_talk_voting_start(user_client):
+    get_default_conference(voting_start=timezone.now() + timedelta(days=1))
+    url = reverse('talk_voting:talks')
+
+    response = user_client.get(url)
+
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/talk_voting/voting_is_closed.html")
+
+
+def test_talk_voting_unavailable_after_talk_voting_end(user_client):
+    get_default_conference(voting_end=timezone.now() - timedelta(days=1))
+    url = reverse('talk_voting:talks')
+
+    response = user_client.get(url)
+
+    assert response.status_code == 200
+    assert template_used(response, "ep19/bs/talk_voting/voting_is_closed.html")
 
 
-@pytest.mark.xfail
-def test_talk_voting_unavailable_without_a_ticket():
-    assert False
+def test_talk_voting_unavailable_without_a_ticket(user_client):
+    get_default_conference()
+    url = reverse('talk_voting:talks')
+
+    response = user_client.get(url)
+
+    assert response.status_code == 200
+    assert not user_client.user.ticket_set.exists()
```

Comment:
I'd go for explicit `is False` here.

Reply:
I think it's a stylistic preference; I prefer `not [...].exists()` as its reads closer to natural language than  `[..].exists() is False`